backend/agents/grading_agent.py

```
@@ -65,23 +65,28 @@ def _create_grading_tools(self) -> List[BaseTool]:
-        def evaluate_learning_objectives(responses: str, objectives: str) -> str:
-            """Evaluate responses against learning objectives"""
-            return f"Evaluating responses against {len(objectives.split(','))} learning objectives"
+        def evaluate_strategic_depth(responses: str, objectives: str) -> str:
+            """Evaluate responses for strategic thinking depth and analytical rigor"""
+            return f"Evaluating strategic depth against {len(objectives.split(','))} learning objectives. Assessing long-term thinking, stakeholder consideration, and critical analysis."
         
         @tool
-        def generate_detailed_feedback(score: int, reasoning: str) -> str:
-            """Generate detailed feedback based on score and reasoning"""
-            return f"Generating feedback for score {score} with reasoning: {reasoning}"
+        def assess_practical_application(responses: str, scene_context: str) -> str:
```

Ruff (0.13.1)

Unused function argument: `responses`

(ARG001)